apps/web/src/db/documents/action.ts

```diff
@@ -0,0 +1,296 @@
+      )
+    )
+}
+
+export async function getDocumentByTable({
+  dataset_id,
+  params,
+}: geDocumentParams) {
+  const { search = '', pageSize = 10, pageIndex = 0 } = params
+  const total = await db
+    .select()
+    .from(DocumentsTable)
+    .orderBy(desc(DocumentsTable.created_at))
+    .where(
+      and(
+        eq(DocumentsTable.dataset_id, dataset_id),
+        eq(DocumentsTable.archived, false),
+        ilike(DocumentsTable.name, `%${search}%`)
+      )
+    )
+  const item: NewDocument[] = await total.slice(
+    pageIndex * pageSize,
+    (pageIndex + 1) * pageSize
+  )
+
+  return { documents: item, total: total?.length }
+}
+
+export async function findApps(ids: string[]) {
+  return await db
+    .select({
+      name: AppsTable.name,
+      icon: AppsTable.icon,
+      uid: AppsTable.api_model_id,
+      app_id: AppsTable.short_id,
+    })
+    .from(AppsTable)
+    .where(
+      and(inArray(AppsTable.api_model_id, ids), eq(AppsTable.archived, false))
+    )
+}
+
+function createEmptyDocument(
+  dataset_id: string,
+  user_id: string,
+  config?: Record<string, any>,
+  attributes?: Record<string, any>
+) {
+  return {
+    config: { splitConfig: config },
+    ...attributes,
+    updated_at: new Date(),
+    dataset_id,
+    created_by: user_id,
+    short_id: nanoid(),
+    status: 1,
+  }
+}
+
+export async function addDocuments(data: newDocumentParams) {
+  try {
+    const { userId } = auth()
+
+    if (!userId) return Promise.resolve([])
+    const { type, documents: _documents, dataset_id, config } = data
+    const isPdf = type === 'pdf'
+
+    let documents = []
+    if (isPdf) {
+      documents = _documents?.reduce((m: DataProps[], cur: DataProps) => {
+        const item = createEmptyDocument(dataset_id, userId, config, cur)
+        // @ts-ignore
+        m.push(item)
+        return m
+      }, [])
+    } else {
+      const ids = _documents?.map((item) => item?.uid)
+      const apps = await findApps(ids)
+      documents = _documents?.reduce((m: any[], cur: any, index) => {
+        const attributes = { ...apps[index], type: 'annotated_data' }
+        const item = createEmptyDocument(dataset_id, userId, config, attributes)
+        m.push(item)
+        return m
+      }, [])
+    }
+    const queue: any[] = []
+
+    documents?.forEach((item: NewDocument) => {
+      const task = db.insert(DocumentsTable).values(item)
+      queue.push(task)
+    })
+    await Promise.all(queue)
+    return { success: true, documents }
+  } catch (error) {
+    console.log('error error:', error)
+  }
+}
+
+export async function getDocumentDetail(id: string) {
+  try {
+    const { userId } = auth()
+    if (!userId) return Promise.resolve([])
+
+    // @ts-ignore
+    const [item]: NewDocument = await db
+      .select()
+      .from(DocumentsTable)
+      .where(eq(DocumentsTable.short_id, id))
+      .limit(1)
+    if (!item) {
+      throw new Error('Document not found')
+    }
```

Comment:
```suggestion
    const [item] = await db
      .select()
      .from(DocumentsTable)
      .where(eq(DocumentsTable.short_id, id))
      .limit(1)
    if (!item) {
      throw new Error('Document not found')
    }
```

可以不指定 type `NewDocument` 吧。